src/Archive/ArchiveEntry.cpp

```diff
@@ -113,11 +117,28 @@ string ArchiveEntry::getName(bool cut_ext)
 	// Sanitize name if it contains the \ character (possible in WAD).
 	string saname = Misc::lumpNameToFileName(name);
 
+	return saname.BeforeFirst('.');
```

Comment:
ok, you're right that wxfilename appears to grab the last dot not the first dot, fixing, but also, question: wouldn't this still mess up . in lump names?